src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Core/CreatableUpdatableResourcesRoot.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Management.V2.Resource.Core.ResourceActions;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Management.V2.Resource.Core
+{
+    /// <summary>
+    /// Implementation of <see cref="ICreatableUpdatableResourcesRoot{IFluentResourceT}">
+    /// </summary>
+    /// <typeparam name="IFluentResourceT">the type of resources in the batch</typeparam>
+    /// <typeparam name="InnerResourceT">the type of inner resource that IFluentResourceT wraps</typeparam>
+    internal class CreatableUpdatableResourcesRoot<IFluentResourceT, InnerResourceT> :
+        CreatableUpdatable<ICreatableUpdatableResourcesRoot<IFluentResourceT>,
+            InnerResourceT,
+            CreatableUpdatableResourcesRoot<IFluentResourceT, InnerResourceT>,
+            IResource,
+            object>,
```

Comment:
i will create an issue in Java repo to track this issue, as the same problem is applicable there as well.